### DISCUSSION TITLE
Kotlin Multiplatform support.

### DISCUSSION THREAD

**Discussion by smyrgeorge** [komapper/komapper#1369] [Ideas]
Hello!

Recently, I started developing a database driver for Kotlin Native.
You can find the implementation here: https://github.com/smyrgeorge/sqlx4k
It supports `PostgreSQL`, `MySQL` and `SQLite`

I would like to integrate it with `komapper`. 

Is there any plan to add support for KMP to the core modules?
Actually, with a bit of guidance I could help on that.

If you are interested, let my know 😀

Thanks a lot

**Comment by nakamura-to** [comment]
Thank you for your interest in Komapper! I appreciate your enthusiasm.

To be honest, I don't have much experience with KMP. However, if it's possible to support KMP without making significant changes to the komapper-core module, I’d be open to considering it.

Could you help me understand what's required to add KMP support?

**Reply by smyrgeorge** [reply]
The core module should be easy, since it's written in pure kotlin and the only dependency is the coroutines package (which also supports kotlin native). In the next days I could try it and provide you a merge request.

It's ok to you?

In that case we can also leave this thread open for further discussion.

**Reply by nakamura-to** [reply]
Thank you! Your merge request is very welcome.

**Comment by smyrgeorge** [comment]
Hello!

I just started to make changes to the build setup.
Here is the branch: https://github.com/smyrgeorge/komapper/tree/feature/kotlin-multiplatform-support

If you want, I can open a PR, so you can track the changes from there.
Let me know

**Reply by nakamura-to** [reply]
Hello! @smyrgeorge 

Thank you for the update.
Yes, please go ahead and open a PR so that I can track the changes from there.
Looking forward to seeing the progress!
